### ISSUE TITLE
The plugin doesn't report the source of synax error

### ISSUE THREAD

**Issue by friend-nicen** [ModyQyW/vite-plugin-stylelint#17]
### Describe the bug 描述问题

The plugin doesn't report the source of synax error .
But ，it's  ok when i use CLI.

不知道来源，就不知道改哪里了。

### Reproduction 复现

This is my console when use plugin.

```
 WARN  19:06:27 [vite] warning: Unexpected empty block (block-no-empty)                                                                                                       19:06:27  
  Plugin: vite:stylelint
````
This is my console when use Cli.
```
G:\PHPproject\eslint>npx stylelint src/app.vue
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

src/app.vue
 17:6  ‼  Unexpected empty block   block-no-empty

```
Cli状态下是有错误出处的。

### System Info 系统信息

```Shell
window10
```

### Used Package Manager 使用的包管理器

yarn

### Validations 核对

- [X] Follow our [Code of Conduct](https://github.com/ModyQyW/.github/blob/main/CODE_OF_CONDUCT.md) 遵循我们的 [行为准则](https://github.com/ModyQyW/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate. 检查是否已经有一个报告相同错误的问题，以避免重复创建。
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead. 这是一个具体的错误。如有疑问，请 打开GitHub 讨论。
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug. 所提供的复现是这个问题的 [最小复现](https://stackoverflow.com/help/minimal-reproducible-example)。

**Comment by ModyQyW** [comment]
https://github.com/ModyQyW/vite-plugin-stylelint/blob/main/src/index.ts#L75-L88

插件调用 `this.warn` 完全符合函数签名。恐怕是 `vite` 内部调整了 this.warn 的处理，所以没有显示行号。插件这边无能为力。

**Comment by friend-nicen** [comment]
```
11:7  error  'c' is assigned a value but never used  no-unused-vars
✖ 1 problem (1 error, 0 warnings)
```

感谢百忙之中解惑。

我尝试了一下eslint的同类型插件，他们的做法是如果不符合规则，他们会输出一串可点击的 字符，然后点击就直接跳转到对应代码位置了。

可否能参考参考。

**Comment by ModyQyW** [comment]
早上建了个 demo 跑了一下，发现 `this.warn` 有正确的文件和行号提示。我感觉可能是插件内代码哪里出了问题，我调试一下

**Comment by friend-nicen** [comment]
好的，辛苦辛苦==。

**Comment by ModyQyW** [comment]
v3.0.6 为插件内部方法增加了 `ctx` 参数，现在应该会正常一些。

我认为这个插件只是让 `stylelint` 能和 `vite` 结合起来使用，并不保证错误信息的格式等，这样能确保插件的纯粹、简单、可维护。不妨看看 <https://stylelint.io/user-guide/usage/options#formatter> 和 <https://stylelint.io/developer-guide/formatters>，换用不同的 `formatter` 或者自己写个 `formatter`，这样应该能有比较好的体验。

**Comment by ModyQyW** [comment]
后续有问题可以回复或者另开一个 ISSUE。谢谢。

**Comment by friend-nicen** [comment]
好的，感谢

**Comment by ModyQyW** [comment]
可以尝试一下 v3.1.0

**Comment by friend-nicen** [comment]
收到🐱‍🐉，辛苦辛苦
